pkg/alertmanager/amcfg.go

```diff
@@ -1017,3 +1020,37 @@ func (whc *webhookConfig) sanitize(amVersion semver.Version, logger log.Logger)
 func (wcc *weChatConfig) sanitize(amVersion semver.Version, logger log.Logger) {
 	wcc.HTTPConfig.sanitize(amVersion, logger)
 }
+
+func (ir *inhibitRule) sanitize(amVersion semver.Version, logger log.Logger) {
+	matchersV2Allowed := amVersion.GTE(semver.MustParse("0.22.0"))
+
+	if usingDeprecatedSyntax := checkNotEmptyMap(ir.SourceMatch, ir.TargetMatch, ir.SourceMatchRE, ir.TargetMatchRE); usingDeprecatedSyntax {
+		msg := "inhibit rule is using a deprecated match syntax which will be removed in future versions"
+		level.Warn(logger).Log("msg", msg)
+	}
+
+	if !matchersV2Allowed && checkNotEmptySlice(ir.SourceMatchers, ir.TargetMatchers) {
+		ir.SourceMatchers = nil
+		ir.TargetMatchers = nil
+		msg := `invalid syntax in 'inhibit_rules', target_matchers and source_matchers matching is supported in AlertManager >= 0.22.0 only - dropping config`
+		level.Warn(logger).Log("msg", msg)
```

Comment:
It might be better to either drop the inhibition rule or fail completely since it might not do what you expect (e.g. it may inhibit more alerts than needed).

Reply:
for the other sanitize functions we just log and drop the config..happy to proceed with that or would it be preferable to error out in this case?